### PR TITLE
Added xtensa kernels to be bazel compatible

### LINKED ISSUE #560: Support accelerated implementations via bazel builds
Currently accelerated implementations are only available using a makefile, but supporting them via bazel will make it significantly easier for some teams to integrate TFLM into other downstream applications that are reliant on bazel for development.

### PR DESCRIPTION
**Pull request by jackagoogle**
This PR introduces a new bazel macro called `tflm_kernel_cc_library` which generates a `cc_library` based on the target architecture. The macro combines the reference sources and headers with the accelerated implementations ensuring that there's only one implementation per kernel.

The `config_setting` is used to declare the CPU type for each of the xtensa cores. This can be extended to other architectures, but this PR starts with xtensa only.

BUG=Fixes #560 and progress towards http://b/194200306

### CODE REVIEW

File: tensorflow/lite/micro/kernels/BUILD [hunk 1]
```diff
@@ -1158,3 +1194,28 @@ cc_test(
         "//tensorflow/lite/micro/testing:micro_test",
     ],
 )
+
+####################################
+# Blaze config settings.
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/BUILD:1199]: bazel

File: tensorflow/lite/micro/build_def.bzl [hunk 2]
```diff
@@ -18,3 +18,64 @@ def generate_cc_arrays(name, src, out, visibility = None):
...
+        srcs = [],
+        hdrs = [],
+        accelerated_srcs = {},
+        accelerated_hdrs = {},
+        deps = [],
+        **kwargs):
+    """Creates a cc_library with the optional accelerated target sources.
+
+    Note:
+      Blaze macros cannot evaluate a select() statement. Therefore, the accelerated_srcs and
```
Comment (advaitjain) [inline tensorflow/lite/micro/build_def.bzl:33]: bazel or starlark

**Review by advaitjain — CHANGES_REQUESTED** [review]
Two small documentation nits.

File: tensorflow/lite/micro/kernels/BUILD (hunk 1 above)
Comment (jackagoogle) [reply tensorflow/lite/micro/kernels/BUILD]: done.

File: tensorflow/lite/micro/build_def.bzl (hunk 2 above)
Comment (jackagoogle) [reply tensorflow/lite/micro/build_def.bzl]: done.

**Review by advaitjain — APPROVED** [review]

**Pull request merged**